tools/cache_creator/cache_creator.cpp

```diff
@@ -295,15 +354,15 @@ llvm::Error RelocatableCacheCreator::addElf(llvm::MemoryBufferRef elfBuffer) {
   entry.dataSize = elfBuffer.getBufferSize();
   entry.hashId = elfLlpcInfoOrErr->cacheHash;
 
-  if (elfLlpcInfoOrErr->llpcMajorVersion != BuildLlpcMajorVersion ||
-      elfLlpcInfoOrErr->llpcMinorVersion != BuildLlpcMinorVersion) {
+  uint32_t llpcMajorVersion = elfLlpcInfoOrErr->llpcVersion.getMajor();
+  uint32_t llpcMinorVersion = elfLlpcInfoOrErr->llpcVersion.getMinor().getValue();
+  if (llpcMajorVersion != BuildLlpcMajorVersion || llpcMinorVersion != BuildLlpcMinorVersion) {
```

Comment:
nit: instead of converting back to two ints, this could construct a VersionTuple from the two constants and use it to perform the check and print version numbers in the error message